test/e2e/receive_test.go

```diff
@@ -715,4 +787,145 @@ func TestReceive(t *testing.T) {
+			 ┌──────────┐
+			 │  Prom    │
+			 └────┬─────┘
+			      │ mTLS
+			 ┌────▼─────┐
+			 │ Ingestor │
+			 └────┬─────┘
+			      │
+			 ┌────▼─────┐
+			 │  Query   │
+			 └──────────┘
+			NB: Made with asciiflow.com - you can copy & paste the above there to modify.
+
+			Self signed certs created using following commands: create.sh
+
+		*/
+
+		t.Parallel()
+		s, err := e2e.NewScenario("e2e_receive_mtls_single_ingestor")
+		testutil.Ok(t, err)
+		t.Cleanup(e2ethanos.CleanScenario(t, s))
+
+		receiveLocalDirSuffix := e2ethanos.ReceiveRelLocalDir("ingestor")
+		testutil.Ok(t, cpyDir("./certs", filepath.Join(s.SharedDir(), receiveLocalDirSuffix)))
+		receiveRWCertsFlags := []string{
+			"--remote-write.server-tls-cert=" + filepath.Join(e2e.ContainerSharedDir, receiveLocalDirSuffix, "e2e_receive_mtls_single_ingestor_server.crt"),
+			"--remote-write.server-tls-key=" + filepath.Join(e2e.ContainerSharedDir, receiveLocalDirSuffix, "testserver.key"),
+			"--remote-write.server-tls-client-ca=" + filepath.Join(e2e.ContainerSharedDir, receiveLocalDirSuffix, "testca.crt"),
+		}
+
+		// Setup Router Ingestor.
+		i, err := e2ethanos.NewIngestingReceiver(s.SharedDir(), "ingestor", receiveRWCertsFlags...)
+		testutil.Ok(t, err)
+		testutil.Ok(t, s.StartAndWaitReady(i))
+
+		promLocalDirSuffix := e2ethanos.PrometheusRelLocalDir("1")
+		testutil.Ok(t, cpyDir("./certs", filepath.Join(s.SharedDir(), promLocalDirSuffix)))
+
+		// Setup Prometheus
+		prom, _, err := e2ethanos.NewPrometheus(s.SharedDir(), "1", defaultPromConfig(
+			"prom1",
+			0,
+			tlsRemoteWriteEntry(
+				e2ethanos.RemoteWriteEndpointHTTPS(i.NetworkEndpoint(8081)),
+				filepath.Join(e2e.ContainerSharedDir, promLocalDirSuffix, "testca.crt"),
+				filepath.Join(e2e.ContainerSharedDir, promLocalDirSuffix, "testclient.key"),
+				filepath.Join(e2e.ContainerSharedDir, promLocalDirSuffix, "e2e_receive_mtls_single_ingestor_client.crt"),
+			), "",
+		), e2ethanos.DefaultPrometheusImage())
+		testutil.Ok(t, err)
+		testutil.Ok(t, s.StartAndWaitReady(prom))
+
+		q, err := e2ethanos.NewQuerierBuilder(s.SharedDir(), "1", []string{i.GRPCNetworkEndpoint()}).Build()
+		testutil.Ok(t, err)
+		testutil.Ok(t, s.StartAndWaitReady(q))
+
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+		t.Cleanup(cancel)
+
+		testutil.Ok(t, q.WaitSumMetricsWithOptions(e2e.Equals(1), []string{"thanos_store_nodes_grpc_connections"}, e2e.WaitMissingMetrics))
+
+		// We expect the data from each Prometheus instance to be replicated twice across our ingesting instances
+		queryAndAssertSeries(t, ctx, q.HTTPEndpoint(), queryUpWithoutInstance, promclient.QueryOptions{
+			Deduplicate: false,
+		}, []model.Metric{
+			{
+				"job":        "myself",
+				"prometheus": "prom1",
+				"receive":    "ingestor",
+				"replica":    "0",
+				"tenant_id":  "default-tenant",
+			},
+		})
+	})
+}
+
+func cpyDir(scrDir, dest string) error {
```

Comment:
What does this function do? A recursive copy from `src` to `dest`? Please add a comment

Reply:
Is there not a library function to do this for us?